### PR TITLE
Added Events section to Courier

### PR DESCRIPTION
**Pull request by Jay-umbr**
Added Events section to Courier with one event - rebuild lucene indexes on deploy, and how to override that event.

### CODE REVIEW

File: Add-ons/UmbracoCourier/Developer/Events.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+---
+versionFrom: 7.0.0
+versionRemoved: 8.0.0
+---
+
+# Events
+
+### RebuildLuceneIndexes
```
Comment (sofietoft) [inline Add-ons/UmbracoCourier/Developer/Events.md]:
```suggestion
## RebuildLuceneIndexes
```

File: Add-ons/UmbracoCourier/Developer/Events.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+By default Lucene/Examine indexes will be rebuilt on Courier deployments - if you wish to override, or stop the indexes from being rebuilt automatically, add a C#/.cs file in App_Code directory on both source and target environments (e.g. overrideScript.cs)
```
Comment (sofietoft) [inline Add-ons/UmbracoCourier/Developer/Events.md]:
```suggestion
By default Lucene/Examine indexes will be rebuilt on Courier deployments. If you wish to override, or stop the indexes from being rebuilt automatically, add a C#/.cs file in App_Code directory on both source and target environments (e.g. overrideScript.cs).
```

File: Add-ons/UmbracoCourier/Developer/Events.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
+---
+versionFrom: 7.0.0
+versionRemoved: 8.0.0
+---
+
+# Events
+
```
Comment (jmayntzhusen) [inline Add-ons/UmbracoCourier/Developer/Events.md]:
```suggestion

Courier has several different events you can hook into and override. They are all placed in the `Umbraco.Courier.DataResolvers.Events` namespace. 

```

**Comment by Jay-umbr** [comment]
Thank you for the feedback - I merged suggested changes, and added a short description to the code snippet @sofietoft

File: Add-ons/UmbracoCourier/Developer/Events.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+Courier has several different events you can hook into and override. They are all placed in the `Umbraco.Courier.DataResolvers.Events` namespace. To override or extend an event, you will need to add an override method of the specific event you want to override, and ensure that is available on both the source and target environment. A quick way to do so is to add a class to your ~/App_Code folder, but in most setups you will probably prefer  to add it somewhere it makes sense to you.
+
+## RebuildLuceneIndexes
+
+By default Lucene/Examine indexes will be rebuilt on Courier deployments - it is possible to override, or stop the indexes from being rebuilt automatically:
```
Comment (sofietoft) [inline Add-ons/UmbracoCourier/Developer/Events.md]:
```suggestion
By default Lucene/Examine indexes will be rebuilt on Courier deployments - it is possible to override, or stop the indexes from being rebuilt automatically.
```

File: Add-ons/UmbracoCourier/Developer/Events.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+
+If you implement a class that extends the event you want your IDE should suggest the class you want to override. Here is the base generated class:
```
Comment (sofietoft) [inline Add-ons/UmbracoCourier/Developer/Events.md]:
```suggestion
When you implement a class that extends the event you want, your IDE should suggest the class you want to override. Here is the base generated class:
```

File: Add-ons/UmbracoCourier/Developer/Events.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+        public override void Execute(ItemIdentifier itemId, SerializableDictionary<string, string> parameters)
+        {
+            // do custom things here
+            base.Execute(itemId, parameters);
+        }
+    }
+}
+```
+
+If you want to disable indexing entirely you can implement this and get rid of the base:
```
Comment (sofietoft) [inline Add-ons/UmbracoCourier/Developer/Events.md]:
```suggestion
If you want to disable indexing entirely you can get rid of the base and implement this:
```

**Pull request merged**